extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -2352,23 +2361,24 @@ private void addHeaderParam(BytecodeCreator invoBuilderEnricher, AssignableResul
             String paramName, ResultHandle headerParamHandle, String paramType, ResultHandle client,
             FieldDescriptor methodGenericTypeField, FieldDescriptor methodParamAnnotationsField, int paramIdx) {
 
-        ResultHandle genericType = invoBuilderEnricher.readArrayValue(
-                invoBuilderEnricher.readStaticField(methodGenericTypeField), paramIdx);
+        BytecodeCreator mc = invoBuilderEnricher.ifNull(headerParamHandle).falseBranch();
+        ResultHandle genericType = mc.readArrayValue(
+                mc.readStaticField(methodGenericTypeField), paramIdx);
 
-        ResultHandle parameterAnnotations = invoBuilderEnricher.readArrayValue(
-                invoBuilderEnricher.readStaticField(methodParamAnnotationsField), paramIdx);
+        ResultHandle parameterAnnotations = mc.readArrayValue(
```

Comment:
with that much of bytecode generation, a name more meaningful than `mc` would be great, maybe `notNullValue`, as above?

Reply:
Sure, done